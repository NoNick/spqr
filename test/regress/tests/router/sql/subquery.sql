\c spqr-console
CREATE SHARDING RULE r1 COLUMN i;
CREATE KEY RANGE kridi1 from 0 to 11 route to sh1;
CREATE KEY RANGE kridi2 from 11 to 21 route to sh2;

\c regress

CREATE TABLE tbl(id int, i int);

INSERT INTO tbl (id, i) VALUES(1, 1);

INSERT INTO tbl (id, i) VALUES(12, 12);
INSERT INTO tbl (id, i) VALUES(13, 12);

delete from tbl
where id in (select id from tbl where i = 12 for update skip locked limit 1) and i = 12 returning *;

DROP TABLE tbl;
\c spqr-console
DROP KEY RANGE ALL;
DROP SHARDING RULE ALL;
