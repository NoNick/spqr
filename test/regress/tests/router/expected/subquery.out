\c spqr-console

		SQPR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE SHARDING RULE r1 COLUMN i;
                          add sharding rule                          
---------------------------------------------------------------------
 created sharding rule r1 for table (*) with columns [i, hash: x->x]
(1 row)

CREATE KEY RANGE kridi1 from 0 to 11 route to sh1;
         add key range          
--------------------------------
 created key range from 0 to 11
(1 row)

CREATE KEY RANGE kridi2 from 11 to 21 route to sh2;
          add key range          
---------------------------------
 created key range from 11 to 21
(1 row)

\c regress
CREATE TABLE tbl(id int, i int);
NOTICE: send query to shard(s) : sh1,sh2
INSERT INTO tbl (id, i) VALUES(1, 1);
NOTICE: send query to shard(s) : sh1
INSERT INTO tbl (id, i) VALUES(12, 12);
NOTICE: send query to shard(s) : sh2
INSERT INTO tbl (id, i) VALUES(13, 12);
NOTICE: send query to shard(s) : sh2
delete from tbl
where id in (select id from tbl where i = 12 for update skip locked limit 1) and i = 12 returning *;
NOTICE: send query to shard(s) : sh2
FIXME: here should be deleted values
DROP TABLE tbl;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SQPR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

DROP SHARDING RULE ALL;
    drop sharding rule    
--------------------------
 dropped sharding rule r1
(1 row)

